### DISCUSSION TITLE
Help! - Problems of connection between two collections with MongoDB, through forms.

### DISCUSSION THREAD

**Discussion by migueljcpinto** [neuefische/questions#667] [Web]
Hi there!!!

I have a problem where I cannot receive submitted dates, using React Date Picker, through a form to insert vacations in the database for each element from another collection. The date data should be stored in a collection just for work dates, which would be connected to another "main" collection of workers. The name os the collections are "nurseWorkDates" and "nurses".

The overall aim is to have a form for three aspects of working dates: vacations, days off and availability to work. For the current User Story I wanted to create the vacation form first, and then add the others. 

I have also tried adding this unified form into a set of Tabs, to then split each component into each tab. 

I can see that there are entries from one array with empty content, and not linked to the other collection.

Honestly, I've already changed the code so much, trying various ways and trying to adapt from other codes I've been researching, that I can no longer find the error. Maybe it's just something poorly worded, one letter off or even a total stupidity of confusion, but I'd ask for a look. 

My actual [PR](https://github.com/MiketheBite/Capstone-project/tree/userStory4). 

Thank you in advance for any help. 🍻

**Comment by F-Kirchhoff** [comment]
One problem is, that you try to get the `id` from the `router.query` in your `SchedulePage.js`. But you cannot access this info, since the route to this subpage does not have a dynamic part with the name `id`. (but you have it in another route though). 
In general, you need to rename your routes and rethink where the different pages belong. Please don't name the folder like "NursePage" or "SchedulePage", since these routes will irritate users when they read the url of the subpage. Also. please only use lowercase characters and separate words with a '-'. You will need to place the "Schedule Page" as a **subpage** of your `[id]` folder in order to receive the dynamic id part of the route.

**Reply by migueljcpinto** [reply]
Thank you very much for answering and helping with this process. I will try to correct and adapt the code in the way you advised.

 Just to be visually sure of what you mean with "You will need to place the "Schedule Page" as a subpage of your [id] folder in order to receive the dynamic id part of the route."  

pages/
├── nurses/
│   ├── [id]/
│   │   ├── index.js ( or nurse-page.js)
│   │   └── schedule.js
│   └── index.js (or nurse-list-page.js)

You mean that, right?

🍻 cheers
